### PR TITLE
Resolve #9885 CoqIDE does not work on Windows

### LINKED ISSUE #9885: CoqIDE does not work on Windows
Since the switch to GTK3, CoqIDE has quite a few issues on Windows:

* It crashes on file open - this can be fixed by providing a compiled scheme file
* Icons don't work - this can be fixed by providing an icon scheme. This has the issue that the complete icon scheme is very large (40MB or so) and that coqtop scans the schare folder for plugins and adding the icons adds about 20.000 system calls to the start of each coqtop for the plugin scanning. If only those icons are copied which ara actually used, it is ok, but I have no idea if it always uses teh same icon size on all systems / settings.
* Syntac highlighing doesn't work
* Even with the file open patch mentioned above, CoqIDE usually crashes on file open. I coudln't find out why - need to debug it.

Otherwise it seems to work - if one can get a fil opened, one can get step through it.

### PR DESCRIPTION
**Pull request by MSoegtropIMC**
- Switch gtksourceview to 3.24.11
- Add appropriate set of icons and some other files GTK3 requires
- Add fix for ocamldebug so that this can be debugged

### CODE REVIEW

File: dev/build/windows/makecoq_mingw.sh [hunk 1]
```diff
@@ -910,14 +911,13 @@ function make_libxml2 {
 
 function make_gtk_sourceview3 {
   # Cygwin packet dependencies: intltool
-  # gtksourceview-2.11.2 requires GTK2
-  # gtksourceview-2.91.9 requires GTK3
-  # => We use gtksourceview-2.11.2 which seems to be the newest GTK2 based one
+  # Note: this is always built from sources cause of a bug in the cygwin delivery.
+  # Just dependcies are only built if we build from sources
```
Comment (vbgl) [inline dev/build/windows/makecoq_mingw.sh]: Typo in “dependcies”.

**Review by vbgl — APPROVED** [review]
Thanks Michael for fixing CoqIDE on Windows!

**Comment by MSoegtropIMC** [comment]
I fixed the type and also the shell lint error => should be ready to go.

**Comment by ejgallego** [comment]
Thanks a lot @MSoegtropIMC ! By the way, we'd like to add Windows CI support to the main `lablgtk3` repository, so the risk of breaking the windows build is minimized. What is the setup do you recommend us to follow?

**Comment by MSoegtropIMC** [comment]
@ejgallego : this will be quite tricky because to see the bug fixed here one has to do UI interactions. I am not sure if GTK has something like an event recording and playback mechanism, so that one could automate UI test cases. There are other options, but the (reasonably robust) methods I know of require proprietary UI interaction reco
rders and players.

**Comment by ejgallego** [comment]
@MSoegtropIMC I guess that as a first step we would be happy ensuring compilation and basic examples work on the right environment, so that's indeed the main question, which win + unix subsystem should we choose for our CI?

**Comment by MSoegtropIMC** [comment]
@ejgallego : I am working on a cygwin/opam/dune based development (and eventually also release) build system for Coq. I would use that. It uses a special cygwin/mingw opam repo, though. I can see if the lablgtk examples work this way. The opam repo I mentioned has binary ocaml packages, so that setup and test should be fairly quick this way.

**Comment by ejgallego** [comment]
Sounds good @MSoegtropIMC , are you coordinating with the OCaml / OPAM windows maintainers of that will be a separate thing?

**Comment by MSoegtropIMC** [comment]
@ejgallego : for me it is a bit hard to tell what the "official" Windows opam repo is. I am using this:

<https://github.com/fdopen/opam-repository-mingw>

which did build Coq and its prerequisites and infrastructure tools out of the box with only minor patches (which are already integrated in this repo). This repo appears to be well maintained. But I don't have the impression that this is the "official" Windows opam repo. Do you have an overview?

**Comment by ejgallego** [comment]
> Do you have an overview?

I think that is as official as one can get today, certainly when Windows support becomes more official in they'll start from there.

**Comment by vbgl** [comment]
Can someone take care of merging this? Otherwise I will.

**Review by ejgallego — APPROVED** [review]

**Comment by Zimmi48** [comment]
@vbgl Now that this PR has been merged, and while waiting for the template polymorphism critical bug fix, I'd recommend to release a second beta, even if it is only for one week, like we did with @maximedenes with 8.7+beta2 because CoqIDE was not usable on Windows in 8.7+beta1. If you agree, I can take care of moving the unreleased changelog entries to the Recent changes chapter, and renaming the milestone and project columns.

**Comment by vbgl** [comment]
A second round of β is a good idea. Your help on the change-log and github stuff is much appreciated. Thanks.

**Pull request merged**